packages/framework/src/utils/env.utils.ts

```diff
@@ -7,3 +7,32 @@ export const getResponse = (): typeof Response => {
 
   return CrossFetchResponse;
 };
+
+export const getBridgeUrl = async (): Promise<string> => {
+  /*
+   * Production, staging, or local environments with bring your own local tunnel
+   * An escape hatch for unknown use-cases.
+   */
+  if (process.env.NOVU_BRIDGE_ORIGIN) {
+    return `${process.env.NOVU_BRIDGE_ORIGIN}/api/novu`;
+  }
+
+  // Vercel preview deployments
+  if (process.env.NEXT_PUBLIC_VERCEL_URL) {
+    return `https://${process.env.NEXT_PUBLIC_VERCEL_URL}/api/novu`;
+  }
+
+  // Local environments
+  try {
+    if (process.env.NODE_ENV === 'development') {
+      const response = await fetch('http://localhost:2022/.well-known/novu');
```

Comment:
This is fine for now. In the original doc, since @novu/framework can't tell where Studio is running, there was a best-effort algorithm to use a fixed set of 5 predefined ports and try all of them until one replies. We can leave it as such and get back to it after the release.